File: credit.c
#include <cs50.h>
#include <stdio.h>

int length(long number);

int main(void)
{
    int temp1;
    int temp2;
    long multiplier = 1;
    int sum = 0;
    long number;
    do
    {
        number = get_long("Number: ");
    }
    while (length(number) != 13 && length(number) != 16 && length(number) != 15);

    for (int i = 0; i < length(number); i++)
    {

        temp1 = ((number / multiplier) % 10);
        multiplier *= 10;
        number = number - temp1;
        temp2 = ((number / multiplier) % 10) * 2;
        number = number - temp2;
        sum += (temp1 + (temp2 % 10) + (temp2 / 10 % 10));
        multiplier *= 10;
    }

    if (sum % 10 == 0)
    {
        // printf("%d\n",length(number));
        if (length(number) == 15 && ((((number / 100000000000000 % 10) * 10) + ((number / 10000000000000) % 10) == 34)))
        {
            printf("AMEX\n");
        }
        else if (length(number) == 15 && ((((number / 100000000000000 % 10) * 10) + ((number / 10000000000000) % 10) == 37)))
        {
            printf("AMEX\n");
        }
        else if (length(number) == 16 && (((number / 1000000000000000 % 10) * 10) + ((number / 100000000000000) % 10) == 51))
        {
            printf("MASTERCARD\n");
        }
        else if (length(number) == 16 && (((number / 1000000000000000 % 10) * 10) + ((number / 100000000000000) % 10) == 52))
        {
            printf("MASTERCARD\n");
        }
        else if (length(number) == 16 && (((number / 1000000000000000 % 10) * 10) + ((number / 100000000000000) % 10) == 53))
        {
            printf("MASTERCARD\n");
        }
        else if (length(number) == 16 && (((number / 1000000000000000 % 10) * 10) + ((number / 100000000000000) % 10) == 54))
        {
            printf("MASTERCARD\n");
        }
        else if (length(number) == 16 && (((number / 1000000000000000 % 10) * 10) + ((number / 100000000000000) % 10) == 55))
        {
            printf("MASTERCARD\n");
        }
        else if (((length(number) == 13 || length(number) == 16) &&
                  (((number / 1000000000000) % 10) == 4 || ((number / 1000000000000000) % 10) == 4)))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

int length(long number)
{
    if (number < 10000000000000)
    {
        return 13;
    }
    if (number < 1000000000000000)
    {
        return 15;
    }
    if (number < 10000000000000000)
    {
        return 16;
    }
    else
    {
        return 1;
    }
}

File: credit.py
from cs50 import get_string


sum1 = 0
cardNumber = get_string("Number: ")
if len(cardNumber) == 16:
    for n in range(len(cardNumber)):
        if n % 2 == 0:
            if int(cardNumber[n]) * 2 >= 10:
                sum1 += int((int(cardNumber[n]) * 2) / 10)
                sum1 += int((int(cardNumber[n]) * 2) % 10)
            else:
                sum1 += int(cardNumber[n]) * 2
        else:
            sum1 += int(cardNumber[n])
elif len(cardNumber) in [13, 15]:
    sum1 += int(cardNumber[0])
    for n in range(1, len(cardNumber)):
        if n % 2 == 1:
            if int(cardNumber[n]) * 2 >= 10:
                sum1 += int((int(cardNumber[n]) * 2) / 10)
                sum1 += int((int(cardNumber[n]) * 2) % 10)
            else:
                sum1 += int(cardNumber[n]) * 2
        else:
            sum1 += int(cardNumber[n])

if ((cardNumber[0] + cardNumber[1]) in ["34", "37"]) and (sum1 % 10 == 0):
    print("AMEX")
elif ((cardNumber[0] + cardNumber[1]) in ["51", "52", "53", "54", "55"]) and (
    sum1 % 10 == 0
):
    print("MASTERCARD")
elif (
    ((cardNumber[0]) in ["4"]) and (sum1 % 10 == 0) and len(cardNumber) in [13, 15, 16]
):
    print("VISA")
else:
    print("INVALID")
